src/contracts/BPool.sol

```diff
@@ -660,35 +643,35 @@ contract BPool is BToken, BMath, IBPool {
    * @param to The address to push the pool tokens to
    * @param amount The amount of pool tokens to push
    */
-  function _pushPoolShare(address to, uint256 amount) internal {
+  function _pushPoolShare(address to, uint256 amount) internal virtual {
     _push(to, amount);
   }
 
   /**
    * @dev Mints an amount of pool tokens.
    * @param amount The amount of pool tokens to mint
    */
-  function _mintPoolShare(uint256 amount) internal {
+  function _mintPoolShare(uint256 amount) internal virtual {
     _mint(address(this), amount);
   }
 
   /**
    * @dev Burns an amount of pool tokens.
    * @param amount The amount of pool tokens to burn
    */
-  function _burnPoolShare(uint256 amount) internal {
+  function _burnPoolShare(uint256 amount) internal virtual {
```

Comment:
why the change?

Reply:
@0xteddybear ?
i believe is in order to be able to mock these methods useful for BTT approach

Reply:
precisely! if the method is virtual, `foundry-smock` will generate methods to call this skipping all checks, and promoting it to an external call, allowing us to assert is has been called.